sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosBulkExecutionOptions.java

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.BulkExecutionThresholds;
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.batch.BatchRequestResponseConstants;
+import com.azure.cosmos.implementation.spark.OperationContextAndListenerTuple;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Duration;
+
+/**
+ * Encapsulates options that can be specified for operations used in Bulk execution.
+ * It can be passed while processing bulk operations.
+ */
+@Beta(value = Beta.SinceVersion.V4_19_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public final class CosmosBulkExecutionOptions {
+    private int maxMicroBatchSize = BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST;
+    private int maxMicroBatchConcurrency = BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_CONCURRENCY;
+    private double maxMicroBatchRetryRate = BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_RETRY_RATE;
+    private double minMicroBatchRetryRate = BatchRequestResponseConstants.DEFAULT_MIN_MICRO_BATCH_RETRY_RATE;
+    private Duration maxMicroBatchInterval = Duration.ofMillis(
+        BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_INTERVAL_IN_MILLISECONDS);
+    private final Object legacyBatchScopedContext;
+    private final BulkExecutionThresholds thresholds;
+    private OperationContextAndListenerTuple operationContextAndListenerTuple;
+
+    /**
+     * Constructor
+     * @param thresholds thresholds
+     */
+    CosmosBulkExecutionOptions(Object legacyBatchScopedContext, BulkExecutionThresholds thresholds) {
+        this.legacyBatchScopedContext = legacyBatchScopedContext;
+        if (thresholds == null) {
+            this.thresholds = new BulkExecutionThresholds();
+        } else {
+            this.thresholds = thresholds;
+        }
+    }
+
+    /**
+     * Constructor
+     * @param thresholds thresholds
+     */
+    @Beta(value = Beta.SinceVersion.V4_19_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosBulkExecutionOptions(BulkExecutionThresholds thresholds) {
```

Comment:
don't we need Cosmos prefix for BulkExecutionThresholds -> CosmosBulkExecutionThresholds ?